Android/src/org/droidplanner/android/maps/providers/baidu_map/BaiduMapFragment.java

```diff
@@ -0,0 +1,928 @@
+                    setupMapUI(getBaiduMap());
+                    break;
+                case com.baidu.mapapi.SDKInitializer.SDK_BROADTCAST_ACTION_STRING_PERMISSION_CHECK_ERROR:
+                    Toast.makeText(getActivity().getApplicationContext(),
+                            "BaiduMap: Keycode error, please check the keycode in AndroidManifest.xml!", Toast.LENGTH_SHORT).show();
+                    break;
+                case com.baidu.mapapi.SDKInitializer.SDK_BROADCAST_ACTION_STRING_NETWORK_ERROR:
+                    Toast.makeText(getActivity().getApplicationContext(),
+                            "BaiduMap: Network error!", Toast.LENGTH_SHORT).show();
+                    break;
+            }
+        }
+    };
+
+    private void notifyToLocatorActivity(BDLocation bdloc) {
+        if (mLocationListener != null) {
+            MyLocationData locData = new MyLocationData.Builder()
+                    .accuracy(bdloc.getRadius())
+                    .direction(100).latitude(bdloc.getLatitude())
+                    .longitude(bdloc.getLongitude()).build();
+
+            // convert BDLocation to android.Location
+            Location aLoc = new Location("BaiduMap");
+            //aLoc.setBearing(); No bearing in baidu map
+            aLoc.setLongitude(bdloc.getLongitude());
+            aLoc.setLatitude(bdloc.getLatitude());
+            mLocationListener.onLocationChanged(aLoc);
+        }
+    }
+
+    private void updateBDMapStatus(BDLocation location) {
+        final BaiduMap map = getBaiduMap();
+        final BaiduMapPrefFragment provider = (BaiduMapPrefFragment)(getProvider().getMapProviderPreferences());
+        final Activity activity = getActivity();
+        final Context context = activity.getApplicationContext();
+
+        int mapType =provider.getMapType(context);
+        map.setMapType(mapType);
+
+        MyLocationData locData = new MyLocationData.Builder()
+                .accuracy(location.getRadius())
+                .direction(100).latitude(location.getLatitude())
+                .longitude(location.getLongitude()).build();
+        map.setMyLocationData(locData);
+
+        LatLng ll = new LatLng(location.getLatitude(), location.getLongitude());
+        MapStatusUpdate u = MapStatusUpdateFactory.newLatLng(ll);
+        map.animateMapStatus(u);
+    }
+
+    public class MyBDLocationListenner implements BDLocationListener {
+		@Override
+        public void onReceiveLocation(BDLocation location) {
+			final BaiduMap map = getBaiduMap();
+			final Activity activity = getActivity();
+			if (map == null || location == null || activity == null)
+				return;
+
+            updateBDMapStatus(location);
+            notifyToLocatorActivity(location);
+        }
+
+        public void onReceivePoi(BDLocation poiLocation) {
+        }
+    }
+
+
+    // FIXME: Ugly, Duplicated lat long titude convert functions, have to be templated
+    private LatLng CoordToLatLang(LatLong coord) {
```

Comment:
@droneboost can you use the methods from the [`DroneHelper` class](https://github.com/DroidPlanner/Tower/blob/develop/Android/src/org/droidplanner/android/utils/DroneHelper.java) instead.